src/main/java/com/templateproject/api/controller/TaskController.java

```diff
@@ -0,0 +1,72 @@
+package com.templateproject.api.controller;
+
+import com.templateproject.api.entity.Task;
+import com.templateproject.api.entity.ToDoList;
+import com.templateproject.api.repository.TaskRepository;
+import com.templateproject.api.repository.ToDoListRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+
+
+@RestController
+public class TaskController {
+
+  private TaskRepository taskRepository;
+  private ToDoListRepository toDoListRepository;
+
+  public TaskController(TaskRepository taskRepository, ToDoListRepository toDoListRepository) {
+    this.taskRepository = taskRepository;
+    this.toDoListRepository = toDoListRepository;
+  }
+
+   @GetMapping("/lists/{listId}/tasks")
+  public List<Task> getAllTasksByListId(@PathVariable Long listId) {
+    Optional<ToDoList> optionalList = toDoListRepository.findById(listId);
+
+    return optionalList.map(ToDoList::getTasks).orElse(null);
+}
+
+
+  @GetMapping("/todo-lists/tasks/{taskid}")
+  public Task getTask(Long taskid) {
+    Optional<Task> task = taskRepository.findById(taskid);
+    return task.orElse(null);
+  }
+
+ @PostMapping("/lists/{listId}/tasks")
+  public @ResponseBody Task createTask(@PathVariable Long listId, @RequestBody Task task) {
+    Optional<ToDoList> optionalList = toDoListRepository.findById(listId);
+    if (optionalList.isPresent()) {
+      ToDoList list = optionalList.get();
+      task.setToDoList(list);
+      return taskRepository.save(task);
+    } else {
+    throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Category not found");
+    }
+  }
+
+
+  @PutMapping("/todo-lists/{id}/tasks/{taskid}")
+  public Task updateTask(@PathVariable(value = "id") Long id, @RequestBody Task task) {
```

Comment:
This method is not working, you are retrieving with the PathVariable only the "id" wildCard (corresponding to list id) and then find a task with it. this method needs to be reshaped